runner.py

```diff
@@ -207,5 +223,48 @@ def main():
                 terminated = False
 
 
+def configure_processes(bot_data_dict):
+    team_pids_map = {}
+
+    for player_index, data in bot_data_dict.items():
+        team = data['team']
+        if not team in team_pids_map:
+            team_pids_map[team] = set()
+        team_pids_map[team].update(data['pids'])
+
+    shared_pids = set()
+    cpu_count = psutil.cpu_count()
+    cpus_per_team = cpu_count // 3
+
+    if len(team_pids_map) >= 2 and cpus_per_team > 0:
+        # Sort into three sets of pids: team 0 exclusives, team 1 exclusives, and shared pids
+        # All pids will be assigned high priority
+        # Team exclusive pids will be bound to a subset of cpus so they can't adversely affect the opposite team.
+
+        for team, team_set in team_pids_map.items():
+            if not shared_pids:
+                shared_pids.update(team_set)
+            else:
+                shared_pids.intersection_update(team_set)
+
+        for team, team_set in team_pids_map.items():
+            team_set -= shared_pids
+
+        for team, team_pids in team_pids_map.items():
+            team_cpu_offset = cpus_per_team * team
+            team_cpus = list(range(cpu_count - cpus_per_team - team_cpu_offset, cpu_count - team_cpu_offset))
+            for pid in team_pids:
+                p = psutil.Process(pid)
+                p.cpu_affinity(team_cpus)  # Restrict the process to run on the cpus assigned to the team
+                p.nice(psutil.HIGH_PRIORITY_CLASS)  # Allow the process to run at high priority
```

Comment:
Wonder if this could lead to more choppy recordings since game capture can be cpu intensive.

Reply:
Yep, that's possible. I'll do some testing with both recording and streaming.

Reply:
Here's some stress testing, I don't think anything got worse: https://www.twitch.tv/videos/234291596?t=00h07m25s

Streaming at 720p which is what I ended up with during the tournament after tinkering.